### PR TITLE
deb: fix importing dbgsym packages with versioned Source field

### PR DESCRIPTION
**Pull request by peat-psuwit**
## Requirements

All new code should be covered with tests, documentation should be updated. CI should pass.

## Description of the Change

dpkg-gencontrol can be called with -v flag which set binary package's
version separated from source version. When this happen, the Source
field will contain version number in addition to source package name.
This tripped Aptly's dbgsym restriction, which check for exact source
package name, which in turn prevents the dbgsym & the whole .changes
file from being imported.

From the git history, it seems like this condition is a leftover from
when Aptly filter dbgsym packages using "*-dbgsym". So, I decided to
remove it. A test case has been added to prevent regression.

## Checklist

- [x] unit-test added (if change is algorithm)
- [ ] functional test added/updated (if change is functional)
- [ ] man page updated (if applicable)
- [ ] bash completion updated (if applicable)
- [ ] documentation updated
- [ ] author name in `AUTHORS`

### CODE REVIEW

**Comment by peat-psuwit** [comment]
There seem to be some PGP failures due to key expiration/revocation.

**Comment by peat-psuwit** [comment]
The commit is rebased. PTAL.

**Comment by mariogrip** [comment]
ping @lbolla

**Review by mariogrip — APPROVED** [review]
LGTM :+1:

File: deb/changes.go [hunk 1]
```diff
@@ -195,11 +195,6 @@ func (c *Changes) PackageQuery() PackageQuery {
 			}
 		}
 
-		ddebQuery = &AndQuery{
```
Comment (lbolla) [inline deb/changes.go:198]: Doesn't removing filtering by `c.Source` potentially select the wrong packages?

Comment (peat-psuwit) [reply deb/changes.go:198]: We already have `<pkg name>-dbgsym` rule for each specific package, and it's not like multiple source packages should produce the same binary package. `binaryQuery` doesn't have `c.Source` rule, and it's constructed essentially the same way as `ddebQuery`.

**Comment by lbolla** [comment]
@peat-psuwit if you rebase this against master I can then merge.

**Comment by peat-psuwit** [comment]
Please take another look. The test failure seems to be external.

**Comment by peat-psuwit** [comment]
@lbolla please take another look.

**Comment by lbolla** [comment]
@neolynx What do you think of this patch? I am not too familiar with this part: I'd appreciate your input before merging.

**Review by lbolla — COMMENTED** [review]
@peat-psuwit Would you mind rebasing this once more? master now has fully functional unittests run on GH Actions, whose successful run is required for each PR. Thanks!

**Comment by neolynx** [comment]
LGTM

**Comment by peat-psuwit** [comment]
Rebased and adapt to a change in test utils. Please have another look.

**Review by randombenj — APPROVED** [review]
LGTM

**Review by lbolla — APPROVED** [review]

**Pull request merged**